### PR TITLE
Boost sampler test coverage

### PR DESCRIPTION
**Pull request by oibe**

### CODE REVIEW

File: src/samplers/ratelimiting_sampler.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ export default class RateLimitingSampler {
         this._logger = logger || new NullLogger();
 
         if (maxTracesPerSecond < 0) {
-            this._logger._error(`maxTracesPerSecond must be greater than 0.0.  Received ${maxTracesPerSecond}`);
+            throw new Error(`maxTracesPerSecond must be greater than 0.0.  Received ${maxTracesPerSecond}`);
             return;
```
Comment (yurishkuro) [inline src/samplers/ratelimiting_sampler.js]: rm return after throw

File: test/all_samplers.js [hunk 2]
```diff
@@ -18,15 +18,84 @@
...
+    });
+
+    describe('ProbabilisticSampler', () => {
+        it ('throws error on out of range sampling rate', () => {
+            expect(() => { new ProbabilisticSampler(2.0); }).to.throw('The sampling rate must be less than 0.0 and greater than 1.0. Received 2');
+        });
+
+        it ('calls is Sampled, and returns false', () => {
+            let sampler = new ProbabilisticSampler(0.0);
+            assert.isNotOk(sampler.isSampled());
```
Comment (yurishkuro) [inline test/all_samplers.js]: not sure what it means to call isSampled without the trace id

File: test/all_samplers.js [hunk 3]
```diff
@@ -18,15 +18,84 @@
...
+        it ('callback is called on close', () => {
+            let sampler = new ProbabilisticSampler(1.0);
+            let callbackCalled = false;
+            let closeCallback = () => {callbackCalled = true; };
+            sampler.close(closeCallback);
+
+            assert.isOk(callbackCalled);
+        });
+
+        it ('close called without callback', () => {
```
Comment (yurishkuro) [inline test/all_samplers.js]: add this to the table driven test instead

File: test/all_samplers.js (hunk 2 above)
Comment (oibe) [inline test/all_samplers.js]: we use math.random(), because how do you implement an efficient less than on a buffer?

File: test/all_samplers.js [hunk 4]
```diff
@@ -25,25 +25,30 @@ import ConstSampler from '../src/samplers/const_sampler.js';
...
+            { useCallback: [true, false], sampler: new ProbabilisticSampler(0.5) },
+            { useCallback: [true, false], sampler: new RateLimitingSampler(2)},
+            { useCallback: [true, false], sampler: new RemoteSampler('some-service-name') }
         ];
 
         _.each(samplers, (o) => {
             it (o.description + 'close calls callback', () => {
-                let callbackCalled = false;
-                let closeCallback = () => { callbackCalled = true; };
+                if (o.useCallback) {
```
Comment (yurishkuro) [inline test/all_samplers.js]: useCallback is always a non-empty array, so this condition is always true

File: test/all_samplers.js [hunk 5]
```diff
@@ -18,15 +18,78 @@
...
+                new RateLimitingSampler(2),
+                new RemoteSampler('some-service-name')
+            ]
+        });
+
+        _.each(samplers, (o) => {
+            it (o.description + 'close calls callback', () => {
+                if (o.useCallback) {
+                    let callbackCalled = false;
+                    let closeCallback = () => { callbackCalled = true; };
```
Comment (yurishkuro) [inline test/all_samplers.js]: a better test is to use a number, increment in the callback, and assert `== 1`

**Pull request merged**